core/src/debuginfo/source_manager.rs

```diff
@@ -78,10 +80,25 @@ pub enum SourceManagerError {
     /// An attempt was made to read content using invalid byte indices
     #[error("attempted to read content out of bounds")]
     InvalidBounds,
-    /// An attempt to load a source file failed due to an I/O error
-    #[cfg(feature = "std")]
-    #[error(transparent)]
-    LoadFailed(#[from] std::io::Error),
+    /// Custom error variant for implementors of the trait.
+    #[error("{error_msg}")]
+    Custom {
```

Comment:
Should we remove `#[non_exhaustive]` over `SourceManagerError` to be consistent with the other errors?